tests/components/fritz/test_config_flow.py

```diff
@@ -0,0 +1,339 @@
+"""Tests for AVM Fritz!Box config flow."""
+from unittest.mock import Mock, PropertyMock, patch
+
+import pytest
+from requests.exceptions import HTTPError
+
+from homeassistant.components.fritz.const import DOMAIN
+from homeassistant.components.ssdp import (
+    ATTR_SSDP_LOCATION,
+    ATTR_UPNP_FRIENDLY_NAME,
+    ATTR_UPNP_UDN,
+)
+from homeassistant.config_entries import SOURCE_REAUTH, SOURCE_SSDP, SOURCE_USER
+from homeassistant.const import CONF_DEVICES, CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import MOCK_CONFIG
+
+from tests.common import MockConfigEntry
+
+MOCK_USER_DATA = MOCK_CONFIG[DOMAIN][CONF_DEVICES][0]
+MOCK_SSDP_DATA = {
+    ATTR_SSDP_LOCATION: "https://fake_host:12345/test",
+    ATTR_UPNP_FRIENDLY_NAME: "fake_name",
+    ATTR_UPNP_UDN: "uuid:only-a-test",
+}
+
+
+@pytest.fixture(name="fritz")
+def fritz_fixture() -> Mock:
+    """Patch libraries."""
+    with patch("homeassistant.components.fritz.common.FritzConnection") as fritz:
+        yield fritz
+
+
+async def test_user_123(hass: HomeAssistantType):
+    """Test starting a flow by user."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection.__init__",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.fritz.common.FritzConnection.device_manager",
+        new_callable=PropertyMock,
+        return_value="test_dev",
+    ):
```

Comment:
```suggestion
    with patch(
        "homeassistant.components.fritz.common.FritzConnection",
    ), patch(
        "homeassistant.components.fritz.common.FritzStatus",
    ), patch(
        "homeassistant.components.fritz.common.FritzHosts",
    ):
```

You may want to create test fixtures to mock each of these though.

Reply:
First test fixed, hopefully